app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt

```diff
@@ -863,6 +880,108 @@ class VaultRepositoryImpl(
             )
     }
 
+    @Suppress("LongMethod")
+    override suspend fun registerFido2Credential(
+        fido2CredentialRequest: Fido2CredentialRequest,
+        selectedCipherView: CipherView,
+        isVerificationSupported: Boolean,
+        checkUser: suspend (CheckUserOptions, UiHint?) -> CheckUserResult,
+    ): Fido2CreateCredentialResult {
+        val userId = activeUserId ?: return Fido2CreateCredentialResult.Error(
+            CreateCredentialUnknownException("Active user is required."),
+        )
+        val clientData = if (fido2CredentialRequest.callingAppInfo.isOriginPopulated()) {
+            ClientData.DefaultWithCustomHash(
+                hash = fido2CredentialRequest.callingAppInfo.getAppSigningSignatureFingerprint(),
+            )
+        } else {
+            ClientData.DefaultWithExtraData(
+                androidPackageName = fido2CredentialRequest
+                    .callingAppInfo
+                    .getAppOrigin(),
+            )
+        }
+        val origin = fido2CredentialRequest.origin
+            ?: fido2CredentialRequest.callingAppInfo.getAppOrigin()
+
+        val ciphersWithCredentials = ciphersStateFlow.value.data
```

Comment:
⛏️ I'd rename it to `ciphersWithFido2Credentials`, a cipher with username/password would also mean it has credentials but not necessarily Fido2 ones.